internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -369,19 +369,16 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, bmh *bmh_v1alpha1.Bar
 
 	for ann, value := range bmh.Labels {
 
-		// Find the `BMH_INSTALL_ENV_LABEL`, get the infraEnv configured in it
+		// Find the `BMH_INFRA_ENV_LABEL`, get the infraEnv configured in it
 		// and copy the ISO Url from the InfraEnv to the BMH resource.
-		if ann == BMH_INSTALL_ENV_LABEL {
+		if ann == BMH_INFRA_ENV_LABEL {
 			infraEnv := &aiv1beta1.InfraEnv{}
-			// TODO: Watch for the InfraEnv resource and do the reconcile
-			// when the required data is there.
-			// https://github.com/openshift/assisted-service/pull/1279/files#r604213425
 			if err := r.Get(ctx, types.NamespacedName{Name: value, Namespace: bmh.Namespace}, infraEnv); err != nil {
 				return reconcileError{client.IgnoreNotFound(err)}
 			}
 
 			if infraEnv.Status.ISODownloadURL == "" {
-				// the image has not been created yet, try later.
+				// the image has not been created yetlater.
 				r.Log.Infof("Image URL for InfraEnv (%s/%s) not available yet. Retrying reconcile for BareMetalHost  %s/%s",
 					infraEnv.Namespace, infraEnv.Name, bmh.Namespace, bmh.Name)
 				return reconcileRequeue{time.Minute}
```

Comment:
Now that there is a watch on InfraEnv, should this still requeue a reconcile?

Reply:
Good catch, it shouldn't!